src/main_gate.rs

```diff
@@ -247,6 +253,20 @@ impl<const T: usize> MainGateConfig<T> {
         region.name_column(|| "q_m", self.q_m);
         region.name_column(|| "rc", self.rc);
     }
+
+    pub fn into_size<const N: usize>(&self) -> Option<MainGateConfig<N>> {
```

Comment:
Because it converts the current config to another config with a different size. 
```rust
let config = MainGateConfig::<5>::default();
let little_config: MainGateConfig<2> = config.info_size();
```

Reply:
we can add comments to explain the purpose here. Something like:
/// This is to re-use the same config (with size N1) for other gadget with a potential different size N2 (with N2 <=N1)